superduperdb/components/listener.py

```diff
@@ -86,7 +86,8 @@ def __post_init__(self):
             else:
                 self.identifier = f'{self.model.identifier}/{self.id_key}'
         self.features = {}
-        if hasattr(self.select, 'features'):
+
+        if 'features' in dir(self.select):
```

Comment:
Ducktyping -- will be refactored here: https://github.com/SuperDuperDB/superduperdb/issues/1282